### DISCUSSION TITLE
Vertical Synchronization of Station Pieces and other Suggested Features

### DISCUSSION THREAD

**Discussion by VonSchittaass** [OpenRCT2/OpenRCT2#17858] [Feature requests]
As an avid Player of the Rollercoaster Tycoon franchise for over 20 years now, I am delighted by how much the OpenRCT2 community has expanded upon the original limitations of the game. The recent transition from .sv6 to .park save format was phenomenally well executed. Recently I've even been experimenting with the new Hybrid Rollercoaster. 

However, ride synchronization still isn't perfect in my opinion. The day the synchronization width expanded from 1 to 6 tiles was surely a great one, but I fell that it is only logical that vertical synchronization be allowed as well, even if that means that the stations have to be right above each other. **( Idea 1 )**

Another feature that could be added is making the "Triple Launch" and a "Reverse Launch" official modes of ride operation. **( Idea 2 )** 

Lastly, I think there should be a way to save prebuilt scenery (with or without paths and stalls). For example someone could make a pantheon style dining hall with stalls and bathrooms and then save it allowing people to place such a cafeteria in their own parks. 
This also mans that large builds will be way easier to do because someone can build and save floor patterns, model columns, and generic roof types. **( Idea 3 )**
If I were to give this save format an extension I would either call it a .ssf (scenery save format) or a .sne (truncated form of .scene)

**Comment by IntelOrca** [comment]
Have you tried the [scenery manager plugin](https://openrct2plugins.org/plugin/MDEwOlJlcG9zaXRvcnkzMTQyNzU2NDQ=/openrct2-scenery-manager)?

This allows you to save whole chunks of scenery and a lot more for re-use.

Ride synchronization could certainly be improved. Ideally you should be able to simply pick and choose the rides to sync with, rather than relying on it working it out from what is nearby.

**Reply by VonSchittaass** [reply]
I have that plugin, and wit works well, but I fell like it's something that should be in the game naturally, I mean there's a scenario editor and a Sandbox for rides.

**Comment by ocalhoun6** [comment]
>However, ride synchronization still isn't perfect in my opinion. The day the synchronization width expanded from 1 to 6 tiles was surely a great one, but I fell that it is only logical that vertical synchronization be allowed as well, even if that means that the stations have to be right above each other. ( Idea 1 )

I think the ultimate goal is to just be able to select the ride (and station) to synchronize to with a drop-down menu in the ride settings. That way, you can easily synchronize a ride to any station in the park.

For now, if you want to synchronize _any_ two tracked rides together, there _is_ a workaround:

1: Build both rides.

2: Using the tile inspector, copy the 1st station piece from one ride

3: Paste that station piece next to the 1st station piece of the other ride, rotating it if necessary

4: The two rides can now be synchronized, regardless of their location -- they can be on top of each other, at opposite ends of the park, anywhere ... doesn't matter

5: You can then use the tile inspector to make the pasted station piece invisible, if you wish

-----

>Lastly, I think there should be a way to save prebuilt scenery (with or without paths and stalls).

I think the eventual plan (in conjunction with a new saved track design format) is to have the saved track design feature to be able to do this, even if there's no ride being saved ... maybe even if there are multiple rides (or stalls) being saved.

For now, as IntelOrca said, there's a plugin that does this.

**Reply by VonSchittaass** [reply]
Thanks for the work around. Sadly, I'd have to do this every time that I build the ride meaning that I can't save vertically synchronized Möbius coasters, which was my original plan. 

Speaking of which, if vertical synchronization becomes possible the tracks won't even have to go in the same direction, since only the fronts of the station have to be aligned.

**Reply by ocalhoun6** [reply]
>the tracks won't even have to go in the same direction

This is already the case somewhat, as long as the first tiles of the station pieces are within 6 tiles side-by-side.

For example, this coaster synchronizes perfectly:
![Forest Frontiers 2022-08-19 12-36-01](https://user-images.githubusercontent.com/26573303/185694641-7d50f2f0-9167-41de-b46a-82dcb1789f0c.png)

**Reply by VonSchittaass** [reply]
I built coasters like those, but I always felt that they weren't very stackable. 
Though I would agree that synchronization overall has come a long way.

**Comment by Basssiiie** [comment]
A triple launch coaster is also already possible with some help of some cheats:
https://www.youtube.com/watch?v=VMsYSsRbe_g

**Reply by ocalhoun6** [reply]
Yeah, but there could be some value in making it an official feature. It would make it more accessible to less experienced players, and it may make more aesthetically pleasing designs possible. (Especially by not requiring any boosters separate from the station.)

Oh, and the current hacky way of doing this only accelerates the train when it's moving forward, while real life triple launch coasters also accelerate the train when it's going backwards through the station. That added bit of realism would be nice to add.
